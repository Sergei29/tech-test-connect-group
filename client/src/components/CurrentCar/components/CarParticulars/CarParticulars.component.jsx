import React from "react";
import PropTypes from "prop-types";
//style:
import { CarParticularsContainer } from "./CarParticulars.style";

const CarParticulars = ({ bodystyles, drivetrain, emissions, passengers }) => {
  const renderBodyStyles = () =>
    bodystyles.length ? (
      <div className="particularsItem">
        <span>Bodystyle: </span>
        <span>
          {bodystyles.length > 1 ? bodystyles.join(", ") : bodystyles[0]}
        </span>
      </div>
    ) : null;

  const renderDrivetrain = () =>
    drivetrain.length ? (
      <div className="particularsItem">
        <span>Drivetrain: </span>
        <span>
          {drivetrain.length > 1 ? drivetrain.join(", ") : drivetrain[0]}
        </span>
      </div>
    ) : null;

  const renderEmissions = () =>
    Object.keys(emissions).length ? (
      <div className="particularsItem">
        <span>{emissions.template} :</span>
        <span>{emissions.value}</span>
      </div>
    ) : null;

  const renderPassengers = () =>
    passengers ? (
      <div className="particularsItem">
        <span>Passengers: </span>
        <span>{passengers}</span>
      </div>
    ) : null;

  return (
    <CarParticularsContainer>
      {renderBodyStyles()}
      {renderDrivetrain()}
      {renderEmissions()}
      {renderPassengers()}
    </CarParticularsContainer>
  );
};

CarParticulars.defaultProps = {
  bodystyles: [],
  drivetrain: [],
  emissions: {},
  passengers: 0,
};

CarParticulars.propTypes = {
  bodystyles: PropTypes.array,
  drivetrain: PropTypes.array,
  emissions: PropTypes.object,
  passengers: PropTypes.number,
};

export default CarParticulars;
