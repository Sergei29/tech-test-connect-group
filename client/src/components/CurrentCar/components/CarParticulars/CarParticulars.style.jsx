import styled from "styled-components";
import { spacing } from "../../../../constants/styleVariables";

export const CarParticularsContainer = styled.div`
  display: grid;
  grid-template-columns: auto;
  text-align: left;
  .particularsItem {
    padding: ${spacing.unit(2)};
    display: grid;
    grid-template-columns: 250px 1fr;
  }
`;
