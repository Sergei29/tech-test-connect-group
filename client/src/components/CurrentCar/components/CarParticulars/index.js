export { default } from "./CarParticulars.component";
