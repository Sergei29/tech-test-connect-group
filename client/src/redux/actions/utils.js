/**
 * @description generates the error message out of received error from api
 * @param {Object} error error received from api
 * @param {String} entity name of the entity: 'carList' or 'carCurrent'
 * @returns {String} aggregated error message
 */
export const getErrorMessage = (error, entity) => {
  let errorMessage = "API call failed.";

  if (error.response && error.response.data) {
    errorMessage = error.response.data;
  } else if (error.message) {
    errorMessage = error.message;
  }

  return `${errorMessage}. ${entity}`;
};
