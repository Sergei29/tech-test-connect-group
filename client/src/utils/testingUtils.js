import React from "react";
import { shallow, mount } from "enzyme";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

const mockStore = configureMockStore([thunk]);

const store = mockStore({
  carList: [
    {
      id: "xe",
      modelYear: "k17",
      url: "/vehicle/xe",
      description:
        "The most advanced, efficient and refined sports saloon that Jaguar has ever produced",
      price: "£30,000",
      media: [{ media: "vehicle", url: "assets/images/xe_k17.jpg" }],
    },
    {
      id: "xf",
      modelYear: "k17",
      url: "/vehicle/xf",
      description:
        "Luxury business saloon with distinctive design, dynamic drive and state-of-the-art technologies.",
      price: "£36,000",
      media: [{ media: "vehicle", url: "assets/images/xf_k17.jpg" }],
    },
  ],
  carCurrent: {
    id: "xe",
    meta: {
      passengers: 5,
      drivetrain: ["AWD", "RWD"],
      bodystyles: ["saloon"],
      emissions: { template: "CO2 Emissions $value g/km", value: 99 },
    },
  },
  error: null,
  loading: false,
});

export const shallowRendererReduxConnected = (ComponentName, props = {}) =>
  shallow(
    <Provider store={store}>
      <BrowserRouter>
        <ComponentName {...props} />
      </BrowserRouter>
    </Provider>
  );

export const mountRendererReduxConnected = (ComponentName, props = {}) =>
  mount(
    <Provider store={store}>
      <BrowserRouter>
        <ComponentName {...props} />
      </BrowserRouter>
    </Provider>
  );

export const shallowRenderer = (ComponentName, props = {}) =>
  shallow(<ComponentName {...props} />);

export const mountRenderer = (ComponentName, props = {}) =>
  mount(
    <BrowserRouter>
      <ComponentName {...props} />
    </BrowserRouter>
  );
