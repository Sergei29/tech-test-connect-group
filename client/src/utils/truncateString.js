export const truncateString = (stringValue, maxLength) => {
  let truncateLocation = stringValue.indexOf(" ", maxLength);
  if (truncateLocation === -1) return stringValue;
  return stringValue.substring(0, truncateLocation) + "...";
};
