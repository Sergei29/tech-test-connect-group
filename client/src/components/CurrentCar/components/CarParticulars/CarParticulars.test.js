import { shallowRenderer } from "../../../../utils/testingUtils";
import CarParticulars from "./CarParticulars.component";

describe("CarParticulars", () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      bodystyles: ["saloon"],
      drivetrain: ["AWD", "RWD"],
      emissions: { template: "CO2 Emissions $value g/km", value: 99 },
      passengers: 5,
    };
    wrapper = shallowRenderer(CarParticulars, props);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
