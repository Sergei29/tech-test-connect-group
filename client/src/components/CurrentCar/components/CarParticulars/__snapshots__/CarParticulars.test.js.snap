// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarParticulars should match the snapshot 1`] = `
<styled.div>
  <div
    className="particularsItem"
  >
    <span>
      Bodystyle: 
    </span>
    <span>
      saloon
    </span>
  </div>
  <div
    className="particularsItem"
  >
    <span>
      Drivetrain: 
    </span>
    <span>
      AWD, RWD
    </span>
  </div>
  <div
    className="particularsItem"
  >
    <span>
      CO2 Emissions $value g/km
       :
    </span>
    <span>
      99
    </span>
  </div>
  <div
    className="particularsItem"
  >
    <span>
      Passengers: 
    </span>
    <span>
      5
    </span>
  </div>
</styled.div>
`;
